modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java

```diff
@@ -742,48 +723,9 @@ private static List<PainlessContextClassInfo> sortClassInfos(
         Set<PainlessContextClassInfo> classExcludes,
         List<PainlessContextClassInfo> classInfos
     ) {
-
         classInfos = new ArrayList<>(classInfos);
-        classInfos.removeIf(
-            v -> "void".equals(v.getName())
-                || "boolean".equals(v.getName())
-                || "byte".equals(v.getName())
-                || "short".equals(v.getName())
-                || "char".equals(v.getName())
-                || "int".equals(v.getName())
-                || "long".equals(v.getName())
-                || "float".equals(v.getName())
-                || "double".equals(v.getName())
-                || "org.elasticsearch.painless.lookup.def".equals(v.getName())
-                || isInternalClass(v.getName())
-                || classExcludes.contains(v)
-        );
-
-        classInfos.sort((c1, c2) -> {
-            String n1 = c1.getName();
-            String n2 = c2.getName();
-            boolean i1 = c1.isImported();
-            boolean i2 = c2.isImported();
-
-            String p1 = n1.substring(0, n1.lastIndexOf('.'));
-            String p2 = n2.substring(0, n2.lastIndexOf('.'));
-
-            int compare = p1.compareTo(p2);
-
-            if (compare == 0) {
-                if (i1 && i2) {
-                    compare = n1.substring(n1.lastIndexOf('.') + 1).compareTo(n2.substring(n2.lastIndexOf('.') + 1));
-                } else if (i1 == false && i2 == false) {
-                    compare = n1.compareTo(n2);
-                } else {
-                    compare = Boolean.compare(i1, i2) * -1;
-                }
-            }
-
-            return compare;
-        });
-
-        return classInfos;
+        classInfos.removeIf(v -> ContextGeneratorCommon.isExcludedClassInfo(v) || classExcludes.contains(v));
```

Comment:
Nit/question: better use a || operator or Predicate.or?

Reply:
`||` is IMO the easiest to read and shortest :) Also, as far as performance is concerned it's probably faster in effectively all cases, `Predicate.or` can at best inline to the code as it is written I'd say.